pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -84,7 +84,13 @@ private void setMergedValue(GenericRow row, String column, @Nullable Object merg
       row.putValue(column, mergedValue);
     } else {
       // if column exists but mapped to a null value then merger result was a null value
-      row.putDefaultNullValue(column, _fieldSpecMap.get(column).getDefaultNullValue());
+      if (_fieldSpecMap.get(column).isSingleValueField()) {
+        row.putDefaultNullValue(column, _fieldSpecMap.get(column).getDefaultNullValue());
+      } else {
+        // multivalue must necessarily use null or MutableSegmentImpl.updateDictionary fails due to typecasting
+        // primitive to array
+        row.putDefaultNullValue(column, null);
```

Comment:
i think this is a backward incompatible change, but it might be ok to change it earlier since the impact of this change is not large. 

the problem is, the default value of pinot multi-value column, is the default value of singleValue data type, which caused the type casting exception. Generally there are two solutions

1. update the type casting code 
2. Update the current **default null value** to `null` for multi-value column, as shown in this diff.

@rohityadav1993 can you provide some insights on why #1 is not a feasible solution.

Reply:
Typecasting is done assuming Object[] type in the code attached in description(stacktrace). null can be typecasted but becuase defaultNullValue are stored as primitive for multi value fields it fails when we persist defualtNullValue when the merger result is null for multi value field.

Updating defaultNullValue behaviour for multi value will be major change and introducing change typecasting will have a larger impact radius.

The fix should be safe as it still populates null vector and slightly urgent as anyone merging null value for partial upsert would be seeing error records and may not be aware if not observing the metrics.